### ISSUE TITLE
Cannot install pre-release versions of locally-built plugin

### ISSUE THREAD

**Issue by michael-brandt-cu** [mitchellh/vagrant#8253]
### Vagrant version
1.9.0, 1.9.1

This behavior is not present in 1.8.7.

### Host operating system
macOS Sierra 10.12.2

### Guest operating system
N/A

### Vagrantfile
N/A

### Debug output

https://gist.github.com/michael-brandt-cu/6ac3b9c91c152414ee0baf86dfd22f10

### Expected behavior
The plugin installs successfully, like this:
```
mbrandt@nsidc-brandt ~/projects/vagrant-nsidc-plugin (master *=)
  > vagrant plugin install ./pkg/vagrant-nsidc-6.1.2.pre2.gem
Installing the './pkg/vagrant-nsidc-6.1.2.pre2.gem' plugin. This can take a few minutes...
Installed the plugin 'vagrant-nsidc (6.1.2.pre2)'!
```

### Actual behavior
The plugin does not install, encountering a Bundler error instead:
```
mbrandt@nsidc-brandt ~/projects/vagrant-nsidc-plugin (master *=)
  > vagrant plugin install ./pkg/vagrant-nsidc-6.1.2.pre2.gem
Installing the './pkg/vagrant-nsidc-6.1.2.pre2.gem' plugin. This can take a few minutes...
Bundler, the underlying system Vagrant uses to install plugins,
reported an error. The error is shown below. These errors are usually
caused by misconfigured plugin installations or transient network
issues. The error from Bundler is:

Unable to resolve dependency: user requested 'vagrant-nsidc (> 0)'
```

### Steps to reproduce
1. develop a Vagrant plugin with its version set to a prerelease version, like `6.1.2.pre2` in the `.gemspec` file
2. build the plugin as a gem: `rake build` creates `./pkg/vagrant-nsidc-6.1.2.pre2.gem`
3. attempt to install the built gem: `vagrant plugin install ./pkg/vagrant-nsidc-6.1.2.pre2.gem`

### References
- GH-8000

-----

I am able to get the installation to work by making this change: [#8252](https://github.com/mitchellh/vagrant/pull/8252/files)

The `gem_version` line was removed in #8000, specifically 6e40d46c502b6819919f51388ae9b56a7469bc6b. I don't know how adding that line back impacts the other changes made in that PR, but it resolves my issue.

**Comment by chrisroberts** [comment]
Fixed via #8252. Thanks!

**Comment by michael-brandt-cu** [comment]
Great, thanks!
